provider/maas/environ.go

```diff
@@ -1051,7 +1051,11 @@ func (environ *maasEnviron) StartInstance(
 		if err != nil {
 			return nil, common.ZoneIndependentError(err)
 		}
-		interfaces, err = maas2NetworkInterfaces(ctx, startedInst, subnetsMap)
+		domains, err := environ.Domains(ctx)
```

Comment:
Don't we need some sort of context as to what machine/device/interface that we need to be finding out the information about Domains?
Looking at something like:
http://maas.jujugui.org/MAAS/#/domain/0

I can see that you can have multiple domains defined in MAAS, and what hosts are in what domain should define which search paths you get. I think this patch works for the "90%" case where you have a single domain in MAAS and you just assume that everything is searchable.
I would have hoped we might have scoped the search information to the spaces that a machine is in, but I'd be satisfied at a Machine level, since there is really only the machine-global dns resolution path.

Do we just assume that all domains being controlled by MAAS are meant to be included in the search path? Is that the logic it uses for the host machines?

Reply:
We are getting from MaaS the same information that MaaS provides to etc/resolve.conf on Xenial. Basically this is what we were getting when we were scraping files.